### PR TITLE
Update Polars to v0.34.2

### PR DESCRIPTION
**Pull request by philss**
What changed: https://github.com/pola-rs/polars/releases/tag/rs-0.34.0

This version, as always, contain a bunch of improvements and bug fixes. We had to update the examples from tests using "sample".

There is also a implicit casting when "clip" is using floats as min or max.

### CODE REVIEW

File: lib/explorer/polars_backend/series.ex [hunk 1]
```diff
@@ -363,7 +363,7 @@ defmodule Explorer.PolarsBackend.Series do
       do: Shared.apply_series(s, :s_clip_integer, [min, max])
 
   def clip(%Series{} = s, min, max),
-    do: Shared.apply_series(s, :s_clip_float, [min * 1.0, max * 1.0])
+    do: Shared.apply_series(cast(s, :float), :s_clip_float, [min * 1.0, max * 1.0])
```
Comment (cigrainger) [inline lib/explorer/polars_backend/series.ex:366]: Nitpick
```suggestion
    do: s |> cast(:float) |> Shared.apply_series(:s_clip_float, [min * 1.0, max * 1.0])
```

File: lib/explorer/data_frame.ex [hunk 2]
```diff
@@ -3997,16 +3997,16 @@ defmodule Explorer.DataFrame do
...
+        year integer [2012, 2013, 2014]
+        country string ["SYRIAN ARAB REPUBLIC", "EGYPT", "AFGHANISTAN"]
+        total integer [12198, 58198, 2675]
+        solid_fuel integer [1, 224, 1194]
+        liquid_fuel integer [7909, 26501, 1393]
+        gas_fuel integer [3265, 24672, 74]
+        cement integer [816, 6800, 14]
+        gas_flaring integer [208, 0, 0]
+        per_capita float [0.61, 0.66, 0.08]
+        bunker_fuels integer [437, 694, 9]
```
Comment (cigrainger) [inline lib/explorer/data_frame.ex:4009]: So their sampling changed even if we use the same seed? 😬

**Review by cigrainger — APPROVED** [review]
❤️

**Review by josevalim — APPROVED** [review]

File: lib/explorer/data_frame.ex (hunk 2 above)
Comment (philss) [reply lib/explorer/data_frame.ex:4009]: Yeah, unfortunately :/

And I tried to "find" the same seed, but I couldn't. The strange part is that if you are using the ":replace" option, it won´t change the results.

**Pull request merged**